src/wallet/coinselection.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_COINSELECTION_H
+#define BITCOIN_COINSELECTION_H
+
+#include "amount.h"
+#include "primitives/transaction.h"
+#include "random.h"
+#include "wallet/wallet.h"
+
+bool SelectCoinsBnB(std::vector<CInputCoin>& utxo_pool, const CAmount& target_value, const CAmount& cost_of_change, std::set<CInputCoin>& out_set,
```

Comment:
This could use a comment saying what the arguments are and what bnb stands for, with maybe a sentence or two describing the algorithm, too.

Reply:
I have added a very long comment describing what this does and the arguments of the function.